src/schemas.jl

```diff
@@ -254,35 +254,79 @@ accepted_field_type(::SchemaVersion, ::Type{Symbol}) = Union{Symbol,String}
 
-For required field `f::F` of `sv`:
+Return first schema violation. For each required field `f::F` of `sv`:
 
 - Define `A = Legolas.accepted_field_type(sv, F)`
 - If `f::T` is present in `ts`, ensure that `T <: A` or else immediately return `f::Symbol => T::DataType`.
 - If `f` isn't present in `ts`, ensure that `Missing <: A` or else immediately return `f::Symbol => missing::Missing`.
 
 Otherwise, return `nothing`.
 
-See also: [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
 """
 find_violation(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
 
 function _find_violation end
 
+"""
+    Legolas.find_violations(ts::Tables.Schema, sv::Legolas.SchemaVersion)
+
+Return vector of all schema violations. For required fields `f::F` of `sv`:
+
+- Define `A = Legolas.accepted_field_type(sv, F)`
+- If `f::T` is present in `ts`, ensure that `T <: A` or else append `f::Symbol => T::DataType` to output vector.
+- If `f` isn't present in `ts`, ensure that `Missing <: A` or else append `f::Symbol => missing::Missing` to output vector.
+
+Otherwise, return `Any[]`.
+
+See also: [`Legolas.find_violation`](@ref), [`Legolas.validate`](@ref), [`Legolas.complies_with`](@ref)
+"""
+find_violations(::Tables.Schema, sv::SchemaVersion) = throw(UnknownSchemaVersionError(sv))
+
+function _find_violations end
+
 """
     Legolas.validate(ts::Tables.Schema, sv::Legolas.SchemaVersion)
 
-Throws a descriptive `ArgumentError` if `!isnothing(find_violation(ts, sv))`,
+Throws a descriptive `ArgumentError` if `!isempty(find_violations(ts, sv))`,
 otherwise return `nothing`.
 
-See also: [`Legolas.find_violation`](@ref), [`Legolas.complies_with`](@ref)
+See also: [`Legolas.find_violations`](@ref), [`Legolas.find_violation`](@ref), [`Legolas.complies_with`](@ref)
 """
 function validate(ts::Tables.Schema, sv::SchemaVersion)
-    result = find_violation(ts, sv)
-    isnothing(result) && return nothing
-    field, violation = result
-    ismissing(violation) && throw(ArgumentError("could not find expected field `$field` in $ts"))
-    expected = getfield(required_fields(sv), field)
-    throw(ArgumentError("field `$field` has unexpected type; expected <:$expected, found $violation"))
+    results = find_violations(ts, sv)
+    isempty(results) && return nothing
+
+    field_err = Any[]
+    type_err = Any[]
+    for result in results
+        field, violation = result
+        if ismissing(violation)
+            push!(field_err, field)
+        else
+            expected = getfield(required_fields(sv), field)
+            push!(type_err, (field, expected, violation))
+        end
+    end
+    err = ""
+    if !isempty(field_err)
+        fields = "`" * join(field_err, "`, `") * "`"
+        err *= string("Could not find expected field", length(field_err) > 1 ? "s " : " ", fields, " in $ts")
```

Comment:
```suggestion
        err *= string("Could not find expected field(s)",  fields, " in $ts")
```

FWIW I'd be happy enough with this and would find it easier to read as well